app/drivers/kscan/kscan_gpio_matrix.c

```diff
@@ -225,6 +225,8 @@ static int kscan_matrix_read(const struct device *dev) {
     struct kscan_matrix_data *data = dev->data;
     const struct kscan_matrix_config *config = dev->config;
 
+    int num_active = 0;
```

Comment:
You would need to store this in the instance data struct to be used across read passes.

Without this, you could limit/ignore different key presses on each read, if a key press earlier in the scanning comes in after the limit is hit.

Reply:
> You would need to store this in the instance data struct to be used across read passes.

I think I don't fully get it right now. How would you reset the counter in this case?